pkg/policies/controlplane/component-factory.go

```diff
@@ -161,6 +163,15 @@ func NewComponentAndOptions(
 			Name:          "Sink",
 			ComponentType: runtime.ComponentTypeSink,
 		}, nil, option, err
+	} else if alerter := componentProto.GetAlerter(); alerter != nil {
+		component, option, err := components.NewAlerterAndOptions(alerter, componentIndex, policyReadAPI, alerterIface)
```

Comment:
Can alerter be provided via Fx just like etcd client, Prometheus registry etc. All other component constructors have a consistent and minimal signature so that we can introduce custom components in the future. Hope we can keep it that way

Reply:
@tanveergill fixed it